### PR TITLE
Clarify behavior of ^ and $ in regex_replace action.

### PR DESCRIPTION
**Pull request by pdewilde**
Generally when using a tool such as GREP, if your regex has a ^ it will match the start of every line, but the default for RE2 is to match the start of the entire file/string unless the m multiline flag is set within the regex.

Added tests for this behavior and documentation so users are not surprised by this behavior.

### CODE REVIEW

File: templates/commands/render_action_regexreplace_test.go [hunk 1]
```diff
@@ -52,6 +52,42 @@ func TestActionRegexReplace(t *testing.T) {
...
+		{
+			name: "override_multiline_true",
+			initContents: map[string]string{
+				"a.txt": "apple banana\nbanana apple\napple apple\n",
+			},
+			rr: &model.RegexReplace{
+				Paths: modelStrings([]string{"."}),
+				Replacements: []*model.RegexReplaceEntry{
+					{
+						Regex: model.String{Val: "(?flags:re m)\n$"},
```
Comment (pdewilde) [inline templates/commands/render_action_regexreplace_test.go:82]: Hmm, this was working before when I had `(?flags:re m)^apple` I wonder why its saying regex not valid now. Was trying to get a more compelling example, but it seems no good deed goes unpunished.

Comment (drevell) [reply templates/commands/render_action_regexreplace_test.go:82]: You'd do `(?m:your_regex)`.

Comment (pdewilde) [reply templates/commands/render_action_regexreplace_test.go:82]: Thanks!

File: README.md [hunk 2]
```diff
@@ -456,6 +456,11 @@ Params:
     readability. Use a non-capturing group (like `(?:abc)|(?:abc)`) if you need
     grouping without capturing.
 
+    Note that by default, RE2 doesn't use multiline mode, so ^ and $ will match
+    the start and end of the entire file, rather than each line. To enter
+    multiline mode you need to set the flag by including this: `(?m:YOUR_REGEX_HERE)`.
+    More information available in RE2 docs.
```
Comment (yolocs) [inline README.md:462]: nit: add a link to the RE2 doc?

**Review by yolocs — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (pdewilde) [reply README.md:462]: There was one in the previous paragraph, do you think another is useful?

Comment (yolocs) [reply README.md:462]: Don't bother then :)

**Review by drevell — APPROVED** [review]

**Pull request merged**